app/javascript/components/record-form/form/subforms/subform-dialog-fields/component.jsx

```diff
@@ -64,8 +65,30 @@ const Component = ({
       }
     }
   }, []);
-
   return fieldsToDisplay.map(subformSectionField => {
+    let calculation_expression = subformSectionField?.calculation?.expression
+    if (calculation_expression) {
+      let count = Object.keys(values).reduce((prev, curr) => {
```

Comment:
It would be better if you iterate the `sum` instead of `values`, since `values` will have all the fields in the form and that is a lot more to iterate, also you can use `isNil` and `has` from `lodash`:

```javascript
Object.keys(calculation_expression.sum).reduce((prev, current) => {
  if (has(values, key) && !isNil(value[key]) && values[key] !== "") {
      return prev + 1;
  }
  return prev;
}, 0);
```

And please move the code to an utility function.